tests/Shared/WorkloadTesting/BuildEnvironment.cs

```diff
@@ -0,0 +1,159 @@
+                    sdkWithWorkloadNotFound?.Invoke(probePath, solutionRoot.FullName);
+                    throw new InvalidOperationException($"Could not find find a sdk with the workload installed at {probePath} computed from solutionRoot={solutionRoot}.");
+                }
+            }
+            else
+            {
+                 string? dotnetPath = Environment.GetEnvironmentVariable("PATH")!
+                    .Split(Path.PathSeparator)
+                    .Select(path => Path.Combine(path, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "dotnet.exe" : "dotnet"))
+                    .FirstOrDefault(File.Exists);
+                if (dotnetPath is null)
+                {
+                    throw new ArgumentException($"Could not find dotnet.exe in PATH={Environment.GetEnvironmentVariable("PATH")}");
+                }
+                sdkForWorkloadPath = Path.GetDirectoryName(dotnetPath)!;
+            }
+
+            BuiltNuGetsPath = Path.Combine(solutionRoot.FullName, "artifacts", "packages", EnvironmentVariables.BuildConfiguration, "Shipping");
+
+            // this is the only difference for local run but out-of-tree
+            if (expectSdkWithWorkload)
+            {
+                TestAssetsPath = Path.Combine(AppContext.BaseDirectory, "testassets");
+            }
+            else
+            {
+                TestAssetsPath = Path.Combine(solutionRoot!.FullName, "tests");
+            }
+            if (!Directory.Exists(TestAssetsPath))
+            {
+                throw new ArgumentException($"Cannot find TestAssetsPath={TestAssetsPath}");
+            }
+        }
+        else
+        {
+            // CI
+            if (string.IsNullOrEmpty(EnvironmentVariables.SdkForWorkloadTestingPath) || !Directory.Exists(EnvironmentVariables.SdkForWorkloadTestingPath))
+            {
+                throw new ArgumentException($"Cannot find 'SDK_FOR_WORKLOAD_TESTING_PATH={EnvironmentVariables.SdkForWorkloadTestingPath}'");
+            }
+            sdkForWorkloadPath = EnvironmentVariables.SdkForWorkloadTestingPath;
+
+            if (string.IsNullOrEmpty(EnvironmentVariables.BuiltNuGetsPath) || !Directory.Exists(EnvironmentVariables.BuiltNuGetsPath))
+            {
+                throw new ArgumentException($"Cannot find 'BUILT_NUGETS_PATH={EnvironmentVariables.BuiltNuGetsPath}' or {BuiltNuGetsPath}");
+            }
+            BuiltNuGetsPath = EnvironmentVariables.BuiltNuGetsPath;
+
+            TestAssetsPath = Path.Combine(AppContext.BaseDirectory, "testassets");
+        }
+
+        if (!string.IsNullOrEmpty(EnvironmentVariables.SdkForWorkloadTestingPath))
+        {
+            // always allow overridding the dotnet used for testing
+            sdkForWorkloadPath = EnvironmentVariables.SdkForWorkloadTestingPath;
+        }
+
+        HasSdkWithWorkload = expectSdkWithWorkload;
+        sdkForWorkloadPath = Path.GetFullPath(sdkForWorkloadPath);
+        DefaultBuildArgs = string.Empty;
```

Comment:
Does this ever get set to anything other than `string.Empty`?

Reply:
No, I can remove this. Though this would come in handy when adding build tests which will be new'ing up projects. Not needed right now though.

Reply:
I tried removing it, and it triggers some other refactoring. Can we keep this in though, as it will be useful when we add more tests?